modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/GridNearTxLocal.java

```diff
@@ -2500,7 +2500,7 @@ private boolean enlistWriteEntry(GridCacheContext cacheCtx,
                                                         keepCacheObjects,
                                                         deserializeBinary,
                                                         false,
-                                                        getRes,
+                                                        null,
```

Comment:
Can you, please, clarify this change.

Reply:
The previous code was wrong but never caused the problem before the change.
Passing the getRes we'll return it to the user and in the getEntry case, the binary value will be deserialized.
But we still want to expect CacheObject at futures post-closure listeners (like on tx commit), but may gain deserialized Integer instead.
So, this just a copy-on-write fix allows us to have futures data unchanged at its listeners.